data-serving/data-service/src/controllers/case.ts

```diff
@@ -357,20 +357,40 @@ export const upsert = async (req: Request, res: Response): Promise<void> => {
  * Handles HTTP DELETE /api/cases.
  */
 export const batchDel = async (req: Request, res: Response): Promise<void> => {
-    Case.deleteMany(
-        {
-            _id: {
-                $in: req.body.caseIds,
+    if (req.body.caseIds !== undefined) {
+        Case.deleteMany(
+            {
+                _id: {
+                    $in: req.body.caseIds,
+                },
             },
-        },
-        (err) => {
-            if (err) {
-                res.status(500).json(err.message);
-                return;
-            }
-            res.status(204).end();
-        },
-    );
+            (err) => {
+                if (err) {
+                    res.status(500).json(err.message);
+                    return;
+                }
+                res.status(204).end();
+            },
+        );
+        return;
+    }
+    if (req.body.query.trim() === '') {
+        res.status(422).json('query must not be empty');
+        return;
+    }
+    const parsedSearch = parseSearchQuery(req.body.query);
+    const queryOpts = parsedSearch.fullTextSearch
```

Comment:
huh, that won't get you keyword searches or mixed FTS/keyword based search queries.
i.e suppose I want to delete all the stuff I imported for india because it was wrong the query will be "india curator:timothe@google.com" but it will actually delete everything for india :-(

Take a look at L51 in this file on how to handle keywords as well.

Reply:
Ah, thanks. Done.

Reply:
we should probably move that to a shared method, can you leave a todo or do it please? thanks

Reply:
Moved to shared method.